### PR TITLE
INT B-20793 Add GSR role and base (QAE) traits in office/admin app

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

As we continue implementation of creating GSR (Government Surveillance Representative) users, we need to set the base permissions the same as QAE. This has also been requested to be placed on a feature flag, so this PR also implements that logic. Nothing too crazy.

This PR does the following:
- adds all relevant information needed for GSR feature flag 
(you can find the merged PR for `milmove-feature-flags` repo [here](https://github.com/transcom/milmove-feature-flags/pull/196))
- implements creating a GSR role using the local sign-in method
- adds GSR checkbox to admin app for creating/updating/viewing office users
- added in functionality to NOT create an okta account when approving requested office users from devlocal
- set default log in route for GSR & base permissions to be the same as QAE
- checking GSR feature flag on sign in, if `gsr_role` is `false` then that GSR user will get the invalid permissions page
- fixing the requested office users form to not be skinny on load by setting static width
- updated all relevant tests

### How to test

1. In your `envrc` and `developmental-features` files, set the GSR role value to `true`
2. FIre up your `server` and `client`
3. Using local sign-in, create a new GSR user
4. Confirm you are directed to the QAE home page where you can search for whatever
5. Navigate the app, confirm you can do all things QAE
6. Log out
7. Set the flags to `false`
8. Refire your `server`
9. Log in as a GSR
10. You should see the invalid permissions screen

## Screenshots
Screenshot 2024-08-21 at 10 25 38 AM

Screenshot 2024-08-21 at 10 25 13 AM

508
Screenshot 2024-08-22 at 10 13 17 AM

### CODE REVIEW

File: pkg/handlers/authentication/devlocal_test.go [hunk 1]
```diff
@@ -131,7 +137,7 @@ func (suite *AuthSuite) TestCreateUserHandlerOffice() {
 		{
 			userType: MultiRoleOfficeUserType,
 			roleTypes: []roles.RoleType{roles.RoleTypeTIO, roles.RoleTypeTOO,
-				roles.RoleTypeServicesCounselor, roles.RoleTypePrimeSimulator},
+				roles.RoleTypeServicesCounselor, roles.RoleTypePrimeSimulator, roles.RoleTypeGSR, roles.RoleTypeHQ, roles.RoleTypeQae, roles.RoleTypeCustomerServiceRepresentative},
```
Comment (traskowskycaci) [inline pkg/handlers/authentication/devlocal_test.go:140]: Bothers me a little bit that GSR and HQ are capitalized but Qae is not - how tough would that be to make a standard either one way or the other? Feel free to tell me to shove it

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
Would you mind updating the AC to reflect what I think you and Dan talked about this morning - moving this to B-20809? Probably good to leave it there but just strike through and note that it's been moved to a downstream

```
Given I am an requesting office user
When I fill out the request form for an office account
Then there will be an option for me to select the GSR role and submit the form successfully
```

Had one ask in there for GSR vs Gsr but up to you on that one, not critical

File: pkg/handlers/authentication/devlocal_test.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/handlers/authentication/devlocal_test.go:140]: I won't tell you to shove it, but I will tell you that I won't change it haha

**Comment by danieljordan-caci** [comment]
> Would you mind updating the AC to reflect what I think you and Dan talked about this morning - moving this to B-20809? Probably good to leave it there but just strike through and note that it's been moved to a downstream
> 
> ```
> Given I am an requesting office user
> When I fill out the request form for an office account
> Then there will be an option for me to select the GSR role and submit the form successfully
> ```
> 
> Had one ask in there for GSR vs Gsr but up to you on that one, not critical

Done!

File: pkg/handlers/authentication/devlocal_test.go (hunk 1 above)
Comment (traskowskycaci) [reply pkg/handlers/authentication/devlocal_test.go:140]: I suppose I can live with it

**Review by traskowskycaci — APPROVED** [review]
Looks good! Poked around and it seemed like I could do all QAE things.

**Review by TevinAdams — CHANGES_REQUESTED** [review]
Reviewed code and wrapping up working through the test case. Missing 508 compliance check screenshot.

**Comment by danieljordan-caci** [comment]
> Reviewed code and wrapping up working through the test case. Missing 508 compliance check screenshot.

508 added

**Review by TevinAdams — APPROVED** [review]

**Pull request merged**